### PR TITLE
AEA-531 fix for default agent loop timeout on contruction with AEABuild

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes
small refactoring over Agent to make tests easier: now it's possible to spin agents main loop manually

## Fixes

set default agent sleep timeout in AEABuilder to reduce cpu utilization

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that code coverage does not decrease.
- [X] I have checked that the documentation about the `aea cli` tool works
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/agent.py [hunk 1]
```diff
@@ -199,14 +199,20 @@ def start(self) -> None:
 
         :return: None
         """
+        self._start_setup()
+        self._run_main_loop()
+
+    def _start_setup(self) -> None:
```
Comment (DavidMinarsch) [inline aea/agent.py:205]: Great idea! Please make sure the doc string below is clean (correct indentation)

File: tests/common/utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    """
+        context manager to measure execution time of code in context
+
+        :return TimeItResult
+
+        example:
+        with timeit_context() as result:
+            do_long_code()
+        print("Long code takes ", result.time_passed)
+    """
```
Comment (DavidMinarsch) [inline tests/common/utils.py:43]:
```suggestion
    """
    Context manager to measure execution time of code in context

    :return TimeItResult

    example:
    with timeit_context() as result:
        do_long_code()
    print("Long code takes ", result.time_passed)
    """
```
Note the indentation, for consistency I suggest changing this and always start block comments on 4 spaces in.

File: tests/test_aeabuilder.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+from aea.crypto.fetchai import FETCHAI
+
+from tests.common.utils import timeit_context
+
+from .conftest import CUR_PATH
+
+
+def test_default_timeout_for_agent():
+    """
+        Tests agents loop sleep timeout
```
Comment (DavidMinarsch) [inline tests/test_aeabuilder.py:32]: See above re indentation

File: tests/test_aeabuilder.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    agent = builder.build()
+    assert agent._timeout == builder.DEFAULT_AGENT_LOOP_TIMEOUT
+
+    with timeit_context() as time_result:
+        agent._spin_main_loop()
+
+    assert time_result.time_passed > builder.DEFAULT_AGENT_LOOP_TIMEOUT
+    time_0_05 = time_result.time_passed
+
+    """ Timeout == 0.001 """
```
Comment (DavidMinarsch) [inline tests/test_aeabuilder.py:52]: I think it might be better to have a test class and multiple test methods for the different test cases.

Comment (solarw) [reply tests/test_aeabuilder.py:52]: @DavidMinarsch 
at the moment three cases are bound with `assert time_0 < time_0_001 and time_0_001 < time_0_05`
probably this check can be removed and just check that `agent._timeout` set on agent correctly, without measuring sleep time.

**Pull request merged**